packages/arb-token-bridge-ui/src/components/common/TransferCountdown.tsx

```diff
@@ -0,0 +1,60 @@
+import { isTeleport } from '@/token-bridge-sdk/teleport'
+import { DepositStatus, MergedTransaction } from '../../state/app/state'
+import {
+  getOrbitDepositDuration,
+  minutesToHumanReadableTime,
+  useTransferDuration
+} from '../../hooks/useTransferDuration'
+import { isNetwork } from '../../util/networks'
+
+/**
+ * Displays a transfer countdown for a deposit, withdrawal, or cctp.
+ *
+ * @param {MergedTransaction} tx - The transaction object.
+ * @param {boolean} firstLegOnly - Teleport has 2 txns, this flag will give us estimate of only 1st tx, else it will give consolidated duration.
+ */
+export function TransferCountdown({
+  tx,
+  firstLegOnly,
+  textAfterTime = ''
+}: {
+  tx: MergedTransaction
+  firstLegOnly?: boolean
+  textAfterTime?: string
+}) {
+  const { isTestnet } = isNetwork(tx.sourceChainId)
+  let { estimatedMinutesLeft } = useTransferDuration(tx)
+
+  if (estimatedMinutesLeft === null) {
```

Comment:
looking through the code in `useTransferDuration`, the only time `estimatedMinutesLeft` would be `null` is for CCTP

When `tx.status === 'Failure'` or loading, `estimatedMinutesLeft` would be `null`

but `Calculating` is not accurately communicating the failure case

Why are other types of tx always non-null? We should also check their tx status?

Reply:
If tx is not pending we wouldn't be showing time in the UI tho

Reply:
i think in that case we should return null from this component

it didn't have a guarding condition specifically and is up to its usage